src/OptiHPLCHandler/empower_detector_module_method.py

```diff
@@ -26,180 +26,296 @@ def xml_compatible(value: Union[str, bool]) -> str:
+                    timeconstant=channel.find("TimeConstant").text,
+                    ratiominimum=channel.find("RatioMinimum").text,
+                    autozerowavelength=channel.find("AutoZeroWavelength").text,
+                    autozeroinjectstart=to_bool(
+                        channel.find("AutoZeroInjectStart").text
+                    ),
+                    autozeroeventorkey=to_bool(channel.find("AutoZeroEventOrKey").text),
+                )
+            )
+            if (
+                "SingleMode" in datamode
+            ):  # Only load first channel if set to single mode
+                break
+        return channels
+
+    @channels.setter
+    def channels(self, value: list[TUVChannel]):
+        if len(value) not in [1, 2]:
+            raise ValueError("Invalid number of channels entered. Expected 1 or 2.")
+
+        channel_settings = [
+            {
+                "datamode": "SingleMode_1A" if len(value) == 1 else "DualModeA_1B",
+                "datarate": (
+                    "SingleDataRate_20A" if len(value) == 1 else "DualDataRate_1B"
+                ),
+                "timeconstant": "0.1000" if len(value) == 1 else "2.0000",
+                "channel": value[0].to_xml(),
+            },
+            {
+                "datamode": "DualModeB_2C",
+                "datarate": "DualDataRate_1B",
+                "timeconstant": "2.0000",
+                "channel": value[0].to_xml() if len(value) == 1 else value[1].to_xml(),
+            },
+        ]
+
+        for i, settings in enumerate(channel_settings):
+            self[f"Channel{chr(65 + i)}"] = (
+                f"<DataRate>{settings['datarate']}</DataRate>"
+                f"<DataMode>{settings['datamode']}</DataMode>"
+                f"<TimeConstant>{settings['timeconstant']}</TimeConstant>"
+                f"{settings['channel']}"
+            )
 
     @property
-    def channel_dict(self) -> dict[str, dict]:
-        list_of_channel_names = ["ChannelA", "ChannelB"]
-        channel_dict = {}
-        for channel_name in list_of_channel_names:
-            channel_xml = f"<{channel_name}>{self[channel_name]}</{channel_name}>"
-            channel = ET.fromstring(channel_xml)
-            wavelength = channel.find(self.wavelength_name).text
-            channel_dict[self.simplified_channel_name(channel_name)] = {
-                self.wavelength_name: wavelength,
-                "Type": "Single",
-                "XML": channel_xml,
-            }
-        return channel_dict
+    def wavelengths(self) -> list[str]:
+        return [channel.wavelength for channel in self.channels]
+
+    @wavelengths.setter
+    def wavelengths(self, value: list[str]):
+        if not all(isinstance(v, (int, str)) for v in value):
+            raise ValueError("Wavelengths must be a list of strings or integers.")
+        self.channels = [TUVChannel(wavelength=wavelength) for wavelength in value]
```

Comment:
Does this work for int, or should it be 
```suggestion
        self.channels = [TUVChannel(wavelength=str(wavelength)) for wavelength in value]
```